File: mobile/app/src/main/java/com/high5/models/user/LoginDataSource.java
package com.high5.models.user;

import com.high5.database.DatabaseController;

import java.io.IOException;

/**
 * Class that handles authentication w/ login credentials and retrieves user information.
 */
public class LoginDataSource {

    public Result<User> login(String username, String password) {

        try {
            User user = DatabaseController.getInstance(null).getDatabase().userDao().getUser();

            //TODO: authenticate with backend and get token from there.
            //TODO: create service that will do the authentication
            if (user == null) {
                user = new User("HELLO");
                DatabaseController.getInstance(null).getDatabase().userDao().createUser(user);
            }
            return new Result.Success<>(user);
        } catch (Exception e) {
            return new Result.Error(new IOException("Error logging in", e));
        }
    }

    public void logout() {
        // TODO: revoke authentication
    }
}

File: mobile/app/src/main/java/com/bdpsolutions/highfive/data/login/source/LoginDataSource.kt
package com.bdpsolutions.highfive.data.login.source

import com.bdpsolutions.highfive.data.login.Result
import com.bdpsolutions.highfive.data.login.model.User

/**
 * Interface for the login/logout functionality. Will allow mocking for unit tests.
 */
interface LoginDataSource {
    fun login(username: String, password: String): Result<User>
    fun logout()
}